fendermint/app/src/cmd/run.rs

```diff
@@ -15,14 +15,45 @@ use fendermint_vm_interpreter::{
     signed::SignedMessageInterpreter,
 };
 use fendermint_vm_resolver::ipld::IpldResolver;
+use fendermint_vm_topdown::proxy::IPCProviderProxy;
+use fendermint_vm_topdown::sync::launch_polling_syncer;
+use fendermint_vm_topdown::{CachedFinalityProvider, Toggle};
 use fvm_shared::address::Address;
+use ipc_provider::config::subnet::{EVMSubnet, SubnetConfig};
+use ipc_provider::IpcProvider;
 use libp2p::identity::secp256k1;
 use libp2p::identity::Keypair;
+use std::sync::Arc;
 use tracing::info;
 
 use crate::cmd::key::read_secret_key;
 use crate::{cmd, options::run::RunArgs, settings::Settings};
 
+fn create_ipc_provider_proxy(settings: &Settings) -> anyhow::Result<IPCProviderProxy> {
+    let topdown_config = settings.ipc.topdown_config()?;
+
+    let url = topdown_config
+        .ipc_provider_url
+        .parse()
+        .context("invalid agent URL")?;
```

Comment:
So, no more agent? 

It would be nice to move this parsing into the `settings` and use a `Url` type instead of `String`, to catch problems early. There is a type in `tendermint_rpc` we already use [here](https://github.com/consensus-shipyard/fendermint/blob/main/fendermint/app/settings/src/lib.rs#L107), perhaps it can do the trick.